src/integrations/kafka/consumer.cpp

```diff
@@ -255,9 +255,15 @@ void Consumer::Check(std::optional<std::chrono::milliseconds> timeout, std::opti
                                          fmt::format("Couldn't save commited offsets: '{}'", RdKafka::err2str(err)));
     }
   } else {
-    if (const auto err = consumer_->assign(last_assignment_); err != RdKafka::ERR_NO_ERROR) {
+    auto throw_consumer_check_failed = [this](const auto err) {
       throw ConsumerCheckFailedException(info_.consumer_name,
                                          fmt::format("Couldn't restore commited offsets: '{}'", RdKafka::err2str(err)));
+    };
+    if (const auto err = consumer_->assign(last_assignment_); err != RdKafka::ERR_NO_ERROR) {
+      throw_consumer_check_failed(err);
+    }
```

Comment:
It doesn't make sense as the `assign` accepts const reference for the partitions. Why does the information need to be fetched again with `position`?

Reply:
Yes, you are right, I messed it up somehow. This check should go above, when `last_assignment_.empty()`. Will update this soon.

Reply:
I agree with Antonio. Maybe @kostasrim you meant the below call of `position` to be done in the other branch of the `fi`? Because I tried that way and it worked!